### PR TITLE
WIP: Query parameter + Example notes

### PR DESCRIPTION
**Pull request by parsonsmatt**
I've added an endpoint which will exercise the query parameter machinery.

I couldn't get the example to build unless I added the `stack.yaml` file which made stack use `..` for the `servant-purescript` library, otherwise, it'd pull an old version and that would cause stuff to break. I added a note to the example's readme to explain that.

The query parameter code compiles fine, though I haven't written a test for it yet. The only issue I see with it is that the Servant `Server` type for a query parameter is `Maybe a`, and the client codegen takes a plain non-`Maybe` value.

### CODE REVIEW

**Comment by eskimor** [comment]
Ok this is awesome! Thank you - I'll check it out tomorrow!

File: examples/central-counter/README.md [hunk 1]
```diff
@@ -10,3 +10,6 @@
 * `stack exec central-counter`
 * Visit http://localhost:8081/index.html in multiple browser windows and click `+` and `-`.
 * Have fun!
+
+Note that this example is setup to work with the `master` branch of `purescript-servant`.
```
Comment (eskimor) [inline examples/central-counter/README.md:14]: I suspect - you mean servant-purescript? Interesting - I have to check this out, this should not be necessary!

**Comment by eskimor** [comment]
Yeah - it should obviously be a Maybe on the clientside too - I'll fix that. Thank you!

**Comment by parsonsmatt** [comment]
Handling `Maybe` argument types may not be trivial, since it looks like the original type is given in the `servant-foreign` definition.

File: examples/central-counter/README.md (hunk 1 above)
Comment (parsonsmatt) [reply examples/central-counter/README.md:14]: derp, yes. If you do `stack init && stack build` then it resolves the wrong version of the `servant-purescript` library. Explicitly pinning it, either in the cabal file or to the current directory is one way of ensuring that it is consistent

**Comment by eskimor** [comment]
I am afraid, I won't get to it today. Tomorrow I should have some time to check out the build issue and how to make query params a Maybe.

**Comment by eskimor** [comment]
I don't really like, letting PRs hang around, but unfortunately I currently have a deadline to meet at 17. November - and I am a bit behind schedule. I'll check it out, as soon as I can.

**Comment by epsilonhalbe** [comment]
Maybe I'll have some time these days and review it.

**Comment by eskimor** [comment]
Well - the PR itself doesn't look bad, it is more the issues @parsonsmatt raised that I have to check out (and fix).

**Comment by parsonsmatt** [comment]
I would like to make an attempt at providing a patch for the `Maybe` query parameters. I suppose we could do that in a separate PR or I could incorporate it into this one?

**Comment by eskimor** [comment]
Great - just push commits to this one!

**Comment by eskimor** [comment]
Hi there - I should find some time now, that [gonimo.com](https://gonimo.com) is online. What's the status on this? Is there any un-pushed work?

**Comment by parsonsmatt** [comment]
Hey, sorry, I wasn't able to get anything done with it. I hit a block where the type was picked up from `servant-foreign`'s instance which resolves query parameters for `a` to `a`, rather than `a` to `Maybe a`.

**Comment by eskimor** [comment]
ok - no problem - I'll take care of that - might take some time, as my whole family is sick at the moment.

**Comment by eskimor** [comment]
I am not going to be able to take care of this any time soon, so if anyone needs query parameters as Maybe values, feel free to pick this up! :-)

**Pull request closed without merging**